File: common/src/main/java/guepardoapps/lucahome/common/controller/INotificationController.java
package guepardoapps.lucahome.common.controller;

import android.graphics.Bitmap;
import android.support.annotation.NonNull;

import guepardoapps.lucahome.common.models.wirelesssocket.WirelessSocket;

public interface INotificationController {
    String NotificationAction = "NotificationAction";

    String WirelessSocketsAll = "WirelessSocketsAll";
    String WirelessSocketSingle = "WirelessSocketSingle";
    String WirelessSocketData = "WirelessSocketData";

    String WirelessSwitchSingle = "WirelessSwitchSingle";
    String WirelessSwitchData = "WirelessSwitchData";

    void CreateSimpleNotification(int notificationId, int smallIcon, @NonNull Class<?> receiverActivity, Bitmap photo, @NonNull String title, @NonNull String body, boolean autoCancelable);

    void CreateCameraNotification(int notificationId, @NonNull Class<?> receiverActivity);

    void CreateWirelessSocketNotification(int notificationId, @NonNull MutableList<WirelessSocket> wirelessSocketList, @NonNull Class<?> receiverClass);

    void CreateWirelessSwitchNotification(int notificationId, MutableList<WirelessSwitch> wirelessSwitchList, @NonNull Class<?> receiverClass);

    void CreateTemperatureNotification(int notificationId, @NonNull Class<?> temperatureActivity, int icon, @NonNull String title, @NonNull String body, boolean autoCancelable);

    void CloseNotification(int notificationId);
}

File: common/src/main/java/guepardoapps/lucahome/common/controller/INotificationController.kt
package guepardoapps.lucahome.common.controller

import android.support.annotation.NonNull
import guepardoapps.lucahome.common.models.common.NotificationContent
import guepardoapps.lucahome.common.models.wirelesssocket.WirelessSocket
import guepardoapps.lucahome.common.models.wirelessswitch.WirelessSwitch

interface INotificationController {
    fun create(@NonNull notificationContent: NotificationContent)

    fun cameraNotification(@NonNull notificationContent: NotificationContent)
    fun wirelessSocketNotification(id: Int, list: MutableList<WirelessSocket>, receiver: Class<*>)
    fun wirelessSwitchNotification(id: Int, list: MutableList<WirelessSwitch>, receiver: Class<*>)

    fun close(id: Int)
}